### DISCUSSION TITLE
What is Proxy object returned by useFetch on client side?

### DISCUSSION THREAD

**Discussion by lukas-schaetzle** [nuxt/nuxt#16668] [Questions]
The following code works without problems on the server side (meaning if I sourround it with a `process.server` check and inspect `localRef` with vue devtools it is the json with a parsed yaml in it):

```vue
<script setup>
import yaml from "js-yaml"

// if (process.server) {
const localRef= ref({})
const { data } = await useFetch(<...>) // the requested endpoint returns a json with a yaml string in it.
// console.log(data.value)
localRef.value = data.value.data     // *
localRef.value.navigation = yaml.load(localRef.value.navigation)     // *
// }
</script>
```

However, this does not work on the client side (so without `process.server` check): In the browser console I always get an error from the yaml package stating that there's a parsing problem (specifically "end of the stream or a document separator is expected"). If I log the return value from `useFetch` I see a strange proxy object `Proxy { <target>: {…}, <handler>: {…} }` which has the expected value under the key `target`. Why is this there? I didn't find anything documented about that behavior.

And what I find even more strange is that the code works fine (`localRef` is the expected parsed yaml an no error) if I replace the last 2 lines (marked with `*`) with `localRef.value = yaml.load(data.value.data.navigation)`.

Can someone help me to undestand what's going on here?

**Comment by sevillaarvin** [comment]
> I see a strange proxy object Proxy { <target>: {…}, <handler>: {…} } which has the expected value under the key target

According to this https://v3.nuxtjs.org/api/composables/use-fetch, the type signature of `data` property is `Ref`, that's why it is a Proxy and has a `value` property.

----

> And what I find even more strange is that the code works fine (localRef is the expected parsed yaml an no error) if I replace the last 2 lines (marked with *) with localRef.value = yaml.load(data.value.data.navigation).

I think this has to do with how `ref` works.

According to https://vuejs.org/api/reactivity-core.html#ref

> If an object is assigned as a ref's value, the object is made deeply reactive with [reactive()](https://vuejs.org/api/reactivity-core.html#reactive). This also means if the object contains nested refs, they will be deeply unwrapped.

References:

https://stackoverflow.com/a/65262638

**Reply by lukas-schaetzle** [reply]
Thanks for the explanation. That makes sense. (I was also new to Vue 3 and didn't know that this proxy object is the format for refs). However, what I still find strange is that the code works fine on the server side.
